### PR TITLE
Move onto awsbox, update urls for ephemeral and dev deployments, fix email delivery for both

### PR DESCRIPTION
**Pull request by lloyd**
This is a big change to our deployment system.  Previously we had a pile of scripts to interact with AWS for deploying development code or ephemeral snapshots.  There also existed this generalized set of scripts broken into a separate standalone project, called awsbox.

With this change we move browserid deployment onto awsbox directly, so we're a proper user of that project, which decreases maintenance work.  Major features of the change:
- new ephemeral deployment setup steps (see docs)
- hacksign.in becomes personatest.org (for ephemeral deployments)
- dev.diresworb.org becomes dev.anosrep.org (for dev deployments)

We might want to wait to merge this until lloyd returns in about a week, dunno.

this will close issue #1394
this will close issue #1046

### CODE REVIEW

File: docs/AWS_DEPLOYMENT.md [hunk 1]
```diff
@@ -124,9 +137,12 @@ You are pushing *from* the local `myotherbranch`, to the remote `master`.
 These things cost money by the hour, not a lot, but money.  So when you want to
 decommission a VM and release your hold on the DNS name, simply:
 
-    $ scripts/deploy.js destroy some_name_i_chose
-    trying to destroy VM for some_name_i_chose.hacksign.in: done
-    trying to remove DNS for some_name_i_chose.hacksign.in: done
+    $ scripts/deploy.js destroy awstest
```
Comment (shane-tomlinson) [inline docs/AWS_DEPLOYMENT.md]: Above, you talk about still using "some_name_i_chose", these should remain consistent.

File: docs/AWS_DEPLOYMENT.md [hunk 2]
```diff
@@ -139,18 +155,20 @@ There are several things that are pre-configured for your pleasure:
...
 ### User accounts
 
-VMs have two pre-configured users, both which you have passphraseless SSH
+VMs have three pre-configured users, all of which you have passphraseless SSH
 access to:
 
   * `ec2-user` is an account with full sudo access.
   * `app` is an account that has no sudo, receives and builds code via git
     pushes, and runs the application servers.
+  * `proxy` is the account the the HTTP reverse proxy that front-ends your server
```
Comment (shane-tomlinson) [inline docs/AWS_DEPLOYMENT.md:170]: Is there a good description of this somewhere?

**Comment by shane-tomlinson** [comment]
@lloyd - I am excited for this.  I need to get my AWS permissions working correctly and the bundle password before I can fully test this.

**Comment by stephendonner** [comment]
+1

File: docs/AWS_DEPLOYMENT.md (hunk 2 above)
Comment (lloyd) [inline docs/AWS_DEPLOYMENT.md:170]: "good"?  not sure, but here's what exists:  https://github.com/lloyd/awsbox/blob/master/doc/SERVER_LAYOUT.md#the-proxy-user

File: docs/AWS_DEPLOYMENT.md (hunk 1 above)
Comment (lloyd) [inline docs/AWS_DEPLOYMENT.md]: +1, fixing with another commit

**Comment by lloyd** [comment]
blocker: upon signup on main site (personatest.org), cookie errors are displayed on google chrome 20/osx.

**Comment by lloyd** [comment]
blocker: dev.anosrep.org delegation has been broken on us.  yay.  https://bugzilla.mozilla.org/762444

blocker: we need SSL certs for dev.anosrep.org, and the organization's preference is to attain these via geotrust: https://bugzilla.mozilla.org/762229

**Comment by lloyd** [comment]
Given review and testing, I'm going to merge this down now with one change.  dev.diresworb.org will remain the URL for ephemeral deployments, but email delivery will be fixed.

In a separate PR I'll update all development urls to be less diresworby and more anosrepy.

**Pull request merged**